src/CodeComb.vNextChina.Hub/EFNodeProvider.cs

```diff
@@ -16,7 +16,7 @@ public EFNodeProvider(TContext db)
         {
             DB = db;
             Nodes = new List<Node>();
-            foreach (var x in DB.Nodes.ToList())
+            foreach (var x in DB.Nodes)

```

Comment:
Because if you don't make it to list, it will use XXXDataReader to read the database, it is a non thread safe method.